setup.py

```diff
@@ -53,13 +53,7 @@
 package_data['cupy'] += cupy_setup_build.prepare_wheel_libs(ctx)
 
 
-if ctx.setup_command in ('dist_info', 'egg_info'):
-    # Extensions are unnecessary for dist_info generation as all sources files
-    # can be enumerated via MANIFEST.in.
-    print('Skipping extensions configuration')
-    ext_modules = []
-else:
-    ext_modules = cupy_setup_build.get_ext_modules(True, ctx)
```

Comment:
This hack exploited pip's implementation-specific behavior to speed up build times, but I noticed this didn't work well with `uv pip`. Since `uv pip` runs the `setup.py` process only once (unlike `pip` which does twice), this hack is no longer necessary. Contributions that frequently build CuPy from source are recommended to use `uv pip -e . --no-build-isolation` instead of `pip`.

xref https://github.com/cupy/cupy/pull/9444#issuecomment-3451853118